### ISSUE TITLE
Improve line parsing

### ISSUE THREAD

**Issue by kreudom** [TUM-LRR/era-gp-sim#151]

**Comment by ErikP0** [comment]
Hi,
your code looks good, although I recommend to pull from master since this feature is behind ~200 commits.
After that, I did some testing, the string literals do not seem to be recognized:
`simucrash "some message"` generates an error. On the architecture validation side, this instruction node object does not contain a child, so I guess this might be a bug on the parser side.
Unfortunately, I could not test if more than 2 macro parameters are possible now;
This also generates errors:
```C++
.macro myMacro, arg, arg2, arg3, arg4
add arg, arg2, x0
add arg2, arg3, arg4
.endm

myMacro x1, x2, x3, x4
```
On debugging the validation of add, I noticed that the object only has 0 or 1 child node, so I suspect another parser bug (may not be caused by this feature though)

**Comment by davschneller** [comment]
(the string parsing is implemented in issue #140)

**Comment by davschneller** [comment]
Also hi,

the code's looking good; just one suggestion:
For better code position annotation, it would be great to also store the position of the parameters in code (which you might be able to calculate now). Also, the error messages could actually right cover the interval where the error is. (now that this is possible)
And, as far as I can see, there might be a tiny problem with escaping '"': is we got the string `"\\"`, it might try to escape `\"`...
Other than that, I noticed nothing, I like it! (still, now we COULD implement multiple labels per line, but this is just a suggestion...)
